Problem6/src/problem6/Problem6.java

```diff
@@ -10,7 +14,35 @@
      * @param args the command line arguments
      */
     public static void main(String[] args) {
-        // TODO code application logic here
+        Scanner scanner = new Scanner(System.in);
+        
+        String input = scanner.nextLine();
+        
+        StringTokenizer tokenizer = new StringTokenizer(input);
+        
+        String finalOutput = "";
+        String currentLine = "";
+        
+        while (tokenizer.hasMoreTokens()) {
+            
+            String token = tokenizer.nextToken();
+            
+            //validate input
+            if (token.length() > 16) {
+                System.out.println("The input includes " + token + ", which is longer than 16 characters.");
+                return;
+            }
+            
+            
+            if ((currentLine + " " + token).length() > 16) {
+                finalOutput += "\n" + currentLine;
```

Comment:
finalOutput is only being updated when the current line exceeds 16 characters.  This means the last line of your output will never be added to finalOutput.